isthmus/src/main/java/io/substrait/isthmus/expression/ExpressionRexConverter.java

```diff
@@ -229,17 +229,17 @@ public RexNode visit(Expression.IntervalYearLiteral expr) throws RuntimeExceptio
         new BigDecimal(expr.years() * 12 + expr.months()), YEAR_MONTH_INTERVAL);
   }
 
-  private static final long MICROS_IN_DAY = TimeUnit.DAYS.toMicros(1);
+  private static final long MILLIS_IN_DAY = TimeUnit.DAYS.toMillis(1);
 
   @Override
   public RexNode visit(Expression.IntervalDayLiteral expr) throws RuntimeException {
+    long milliseconds =
+        expr.precision() > 3
+            ? (expr.subseconds() / (int) Math.pow(10, expr.precision() - 3))
+            : (expr.subseconds() * (int) Math.pow(10, 3 - expr.precision()));
```

Comment:
These two branches produce the same number don't they?

Reply:
Yes, but with integer operations